realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -405,6 +408,198 @@ public void equalTo_nonLatinCharacters() {
     }
 
     @Test
+    public void in_string() {
+        populateTestRealm(realm, 200);
+        try {
+            realm.where(AllTypes.class).in(AllTypes.FIELD_STRING, (String[]) null).findAll();
+        } catch (IllegalArgumentException ignored) {
+        }
+        try {
+            realm.where(AllTypes.class).in(AllTypes.FIELD_STRING, new String[] {}).findAll();
+        } catch (IllegalArgumentException ignored) {
+        }
+        RealmResults<AllTypes> resultList = realm.where(AllTypes.class).in(AllTypes.FIELD_STRING, new String[] {"test data 15"}).findAll();
+        assertEquals(1, resultList.size());
+        resultList = realm.where(AllTypes.class).in(AllTypes.FIELD_STRING, new String[] {"test data 15", "test data 117", "test data 30", "test data 199"}).findAll();
+        assertEquals(4, resultList.size());
+        resultList = realm.where(AllTypes.class).in(AllTypes.FIELD_STRING, new String[] {"TEST data 15", "TEST data 117", "test DATA 30", "test DATA 199"}, Case.INSENSITIVE).findAll();
+        assertEquals(4, resultList.size());
+        resultList = realm.where(AllTypes.class).not().in(AllTypes.FIELD_STRING, new String[] {"TEST data 15", "TEST data 117", "test DATA 30", "test DATA 199"}, Case.INSENSITIVE).findAll();
+        assertEquals(196, resultList.size());
+        resultList = realm.where(AllTypes.class).not().in(AllTypes.FIELD_STRING, new String[] { "TEST data 15", "TEST data 117", "test DATA 30", "test DATA 199"}, Case.INSENSITIVE).findAll();
```

Comment:
need to test `null` in the list as well. also needs to test it throws when the given field is not nullable.